kafka-connect-aws-s3/src/test/scala/io/lenses/streamreactor/connect/aws/s3/source/files/DateOrderingBatchListerTest.scala

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2017-2023 Lenses.io Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.aws.s3.source.files
+
+import cats.implicits._
+import io.lenses.streamreactor.connect.aws.s3.model.location.RemoteS3PathLocation
+import io.lenses.streamreactor.connect.aws.s3.model.location.RemoteS3RootLocation
+import io.lenses.streamreactor.connect.aws.s3.storage.FileListError
+import io.lenses.streamreactor.connect.aws.s3.storage.FileMetadata
+import io.lenses.streamreactor.connect.aws.s3.storage.ListResponse
+import io.lenses.streamreactor.connect.aws.s3.storage.StorageInterface
+import org.mockito.ArgumentMatchersSugar.any
+import org.mockito.ArgumentMatchersSugar._
+import org.mockito.MockitoSugar
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.BeforeAndAfter
+import org.scalatest.EitherValues
+import org.scalatest.OptionValues
+import software.amazon.awssdk.services.s3.model.S3Object
+
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+
+class DateOrderingBatchListerTest
+    extends AnyFlatSpec
+    with Matchers
+    with MockitoSugar
+    with BeforeAndAfter
+    with OptionValues
+    with EitherValues {
+
+  private val pathLocation     = RemoteS3RootLocation("bucket:path").toPath()
+  private val storageInterface = mock[StorageInterface]
+
+  private val listerFn = DateOrderingBatchLister.listBatch(storageInterface, pathLocation, 10) _
```

Comment:
Can we add the type annotation to make it easier to read this line without looking at listBatch